build_pros.cmake

```diff
@@ -211,9 +211,9 @@ if(NOT WIN32 AND NOT CYGWIN)
         )
 
         cmake_push_check_state()
-        set(CMAKE_REQUIRED_FLAGS "-fvisibility=hidden -Werror")
         check_c_source_compiles("__attribute__((visibility(\"default\"),externally_visible)) int foo(void);
-                                 int foo(void) { return 0; }" HAVE_EXTERNAL_VISIBILITY)
+                                 int foo (void) { return 0; }
```

Comment:
If you move this above just after the main visibility check like I did, you won't
need the extra push/pop, and it will be always checked with the correct cflags.

Reply:
Done! :+1: